lightning/src/ln/channel.rs

```diff
@@ -5481,17 +5481,21 @@ impl<SP: Deref> Channel<SP> where
 		// If we're past (or at) the AwaitingChannelReady stage on an outbound channel, try to
 		// (re-)broadcast the funding transaction as we may have declined to broadcast it when we
 		// first received the funding_signed.
-		let mut funding_broadcastable =
+		let mut funding_broadcastable = None;
+		if let Some(funding_transaction) = &self.context.funding_transaction {
 			if self.context.is_outbound() &&
 				(matches!(self.context.channel_state, ChannelState::AwaitingChannelReady(flags) if !flags.is_set(AwaitingChannelReadyFlags::WAITING_FOR_BATCH)) ||
 				matches!(self.context.channel_state, ChannelState::ChannelReady(_)))
 			{
-				self.context.funding_transaction.take()
-			} else { None };
-		// That said, if the funding transaction is already confirmed (ie we're active with a
-		// minimum_depth over 0) don't bother re-broadcasting the confirmed funding tx.
-		if matches!(self.context.channel_state, ChannelState::ChannelReady(_)) && self.context.minimum_depth != Some(0) {
-			funding_broadcastable = None;
+				// Broadcast only if not yet confirmed
+				if self.context.get_funding_tx_confirmation_height().is_none() || self.context.get_funding_tx_confirmed_in().is_none() {
```

Comment:
Do we check both because `funding_tx_confirmed_in` is not reset during a re-org? Should we only check `funding_tx_confirmation_height`?

Reply:
It both are set, then it's confirmed for sure. But right, it should be enough to check `funding_tx_confirmation_height`